### PR TITLE
Pipeline for creating testing data

### PR DESCRIPTION
**Pull request by chadlagore**
:hourglass: **Status**: _Ready for review_
:tickets: **Ticket(s)**: #40 #37 

## :construction_worker: Changes

+ Incorporated @CoffeeTableEspresso's function (renamed/enforced pep8 on it).
    + Changed the function specification to take `samples_per_observation`.
    + Changed the function specification to take `left_delim` and `right_delim`.
+ Included `extract_observation` specification for @CoffeeTableEspresso to implement. 

**TODO: Fix docker setup so no one has any trouble running this. To run it, just pip install numpy/scipy**.

## :flashlight: Testing Instructions

```
./get-transcript.sh obgTxz-x33Q
python3 pipeline.py obgTxz-x33Q

./get-transcript.sh obgTxz-x33Q
python3 pipeline.py Xz3btMhdQ6Y
```

Read logs, inspect output, etc.

To turn off debugging noise, change `logging.basicConfig(level=logging.DEBUG)` to `logging.basicConfig(level=logging.INFO)` in `extract_labels.py`.

### CODE REVIEW

File: collector/extract_labels.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+def update_current_speaker(current, text, re_delim):
+    """Returns current speaker based on text, previous current speaker,
+    deliminator, and discard
+    """
+    try:
+        return re_delim.findall(text)[-1]
+    except IndexError:
+        return current
```
Comment (chadlagore) [inline collector/extract_labels.py]: This file is @CoffeeTableEspresso's code, I just changed the delimiters, and set up our `samples_per_observation` approach. Inspect the log output to make sure it looks right.

File: collector/extract_observations.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            (row).
+        sample_rate (int): The audio sample rate.
+
+    Returns:
+        np.ndarray: An array with num_samples / samples_per_observation
+            rows and enough columns to evenly distrbute the audio data
+            throughout the table.
+    """
+    # Stub.
+    return np.zeros((0, 0))
```
Comment (chadlagore) [inline collector/extract_observations.py]: @CoffeeTableEspresso this is your new baby. Feel free to change the spec/arguments all you like :tada:.

File: collector/extract_labels.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    if len(speakers) > 1:
+        return False
+    if not on_start and speakers:
+        return False
+    if discard:
+        return False
+    return True
+
+
+def update_current_speaker(current, text, re_delim):
```
Comment (iKevinY) [inline collector/extract_labels.py]: Naming nitpick: `update_` implies that the function is performing some mutation; perhaps something like `get_current_speaker` is more appropriate.

File: collector/pipeline.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+from extract_labels import extract_labels
+from extract_observations import extract_observations
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+# Hyper paramters.
```
Comment (iKevinY) [inline collector/pipeline.py]: `paramters` → `parameters`

File: collector/pipeline.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+SAMPLES_PER_OBSERVATION = 500
+
+
+def build(sample_id):
+    """Produces a training datset from YouTube audio and
```
Comment (iKevinY) [inline collector/pipeline.py]: `datset` → `dataset`

File: collector/extract_labels.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    re_delim = re.compile(pattern)
+
+    # List of all possible speakers.
+    speakers = set()
+    for text in texts:
+        for speaker in re_delim.findall(text):
+            speakers.add(speaker)
+
+    # Map speakers to integers - use OrderedDict so the output
+    # is determinant.
```
Comment (iKevinY) [inline collector/extract_labels.py]: `determinant` → `deterministic`

**Pull request merged**